### ISSUE TITLE
Will you continue to develop the PBA project?

### ISSUE THREAD

**Issue by Kerrung** [acomminos/wine-pba#49]
Hi! Thank you very much for creating your PBA project! Your PBA project is important in those games where DXVK can not work. For example, if the game simultaneously uses a mixture from different APIs, such as d3d9, d3d10 and d3d11, then using DXVK makes it impossible to run such games, but it's not a problem for your PBA.
For example, WoW uses simultaneously pieces d3d9, d3d10 and d3d11 at the same time, and therefore WoW can not work if DXVK is used.
Developer of DXVK said that in the near future does not plan to add support of d3d10. Therefore, your project is still relevant for many Wine users.
But, will you continue to develop the PBA project? Unfortunately, we no longer see any your activity in the development of the PBA, there are no more new commits, and you do not respond to new bug reports.
I would like to see your answer on this issue in order to understand what the situation is at the moment.

Thank you

**Comment by gverm** [comment]
What is the point of this issue? It's only been 12 days, please have some patience. I'm pretty sure this is not a full time job for him. He will release more changes once he has them. Making an issue to ask about the status of a project after only 12days of inactiveness is not necessary.

**Comment by Kerrung** [comment]
> @gverm:  What is the point of this issue? It's only been 12 days, please have some patience. I'm pretty sure this is not a full time job for him. He will release more changes once he has them. Making an issue to ask about the status of a project after only 12days of inactiveness is not necessary.

Earlier, I saw about three new commits a day every day. But now there is not even a rebase for Wine 3.4, but Wine 3.5 will be released soon.

**Comment by Enverex** [comment]
It worked with 3.4 at least at one point because I have a successful package build:

wine-staging-pba-git-3.4.r0.g495f9db4+wine.3.4.r0.gafe4f54bb4-1-x86_64.pkg.tar.gz

**Comment by Firerat** [comment]
@Kerrung rebasing the current patches is fairly trivial, I verbosely burbled about that [on a wiki page](https://github.com/Firerat/wine-pba/wiki/patch-rebase-and-conflicts)

@Enverex I can look at that, PBA dosn't really touch the staging csmt code ( only one trivial patch to the csmt Makefile.in )

**Comment by Firerat** [comment]
@gloriouseggroll has already fixed this issue with csmt
https://github.com/GloriousEggroll/wine-pba

**Comment by Enverex** [comment]
That fixes that one, but it still runs into an issue further down the line (although it got to patch 8 this time).

```
  -> Applying PBA patch /var/tmp/makepkg/wine-staging-pba-git/src/wine-pba/patches/0008-wined3d-Add-DISABLE_PBA-envvar-some-PBA-cleanup.patch
patching file dlls/wined3d/buffer.c
Hunk #1 succeeded at 1601 (offset -2 lines).
patching file dlls/wined3d/buffer_heap.c
patching file dlls/wined3d/device.c
Hunk #1 succeeded at 844 (offset 7 lines).
Hunk #2 succeeded at 873 (offset 7 lines).
patching file dlls/wined3d/query.c
patching file dlls/wined3d/wined3d_private.h
Hunk #2 FAILED at 2943.
1 out of 2 hunks FAILED -- saving rejects to file dlls/wined3d/wined3d_private.h.rej
```

Looks like the section this patch removes doesn't actually exist anymore so likely an easy fix.

**Comment by Firerat** [comment]
check the wiki, it is the issue I cover in that
dosn't have the actual answer  but has some very heavy hints at it

also check https://github.com/GloriousEggroll/wine-pba  it is probably already fix there

**Comment by Enverex** [comment]
Sure, but I was documenting what the issues are as they stand now so that they can be fixed here. Your Wiki entry is just a guide on general merge issues when people are trying to fix them themselves.

The last issue I mentioned though is still live on GloriousEggroll's fork anyway.

**Comment by Firerat** [comment]
well, I ran into no such issues, but I used `git am --3way`  

[PBA-for_staging_5876a3f7.tar.gz](https://github.com/acomminos/wine-pba/files/1856585/PBA-for_staging_5876a3f7.tar.gz)
Note, patch 0010 in that is optional

**Comment by goldpaw** [comment]
Or for the lazy, here's a single patch file you can patch the most recent wine-staging 3.4 with. Created after the CSMT_Helper changes and all that. Patched fine enough for me, I'm building wine with it as we speak. 

[wine-staging-pba.zip](https://github.com/acomminos/wine-pba/files/1857946/wine-staging-pba.zip)

**Comment by goldpaw** [comment]
Well your hobby has from doubled to tripled the framerate in games I had completely given up on, enabling me to finally play some of my favorite games again! So a huge thank you from me for all the work you've done so far. You rock! :)

**Comment by ryanmusante** [comment]
Could you give examples of games you see this increase in? I'd like to add to my list. I also use PBA.

**Comment by goldpaw** [comment]
World of Warcraft and Path of Exile.

**Comment by Saroufim** [comment]
Path of Exile seems to lose FPS on my system with PBA.

**Comment by goldpaw** [comment]
@Saroufim we're talking about Path of Exile 64bit with DX11, right? Because it was completely unplayable for me with the 32bit DX9 client. I'm using a tiny patch to get it to work with DX11.

**Comment by Saroufim** [comment]
That must be it because it doesn't work at all with DX11 on my computer. I'm forced to use DX9

**Comment by goldpaw** [comment]
yeah, it was unplayable for me with Dx9. constant stuttering, poor fps, etc. If it helps, here's the patch I'm using. I apply it to the wine folder before staging and pba. Then I build wine, and run Path of Exile in 64bit mode with some launch options to preload spell effects and improve garbage collection to avoid out of memory crashes: 

`wine 'PathOfExile_x64.exe' --garbage-generation 1 --waitforpreload`

[poe-fix.zip](https://github.com/acomminos/wine-pba/files/1887506/poe-fix.zip)

**Comment by goldpaw** [comment]
@Saroufim and turn of engine multithreading in Path of Exile. Leave that to CSMT and wine, there's a notable framerate difference.
